extend-generics/test.ooc

```diff
@@ -12,22 +12,22 @@ extend ExampleClass<T> {
 		}
 		result
 	}
-	typeName: func -> String { T name toString() }
+	typeName: func -> String { T name }
 }
 
 myClass := ExampleClass<Int> new(42)
 myName := myClass typeName()
 myInt := myClass someMethod()
-raise(typeName != "Int", "typeName is not Int but %s" format(myName))
+raise(myName != "Int", "typeName is not Int but %s" format(myName))
 raise(myInt != 42, "someMethod did not return 42 but %d" format(myInt))
 
 myClass free()
 myName free()
 
-myClass = ExampleClass<String> new(null)
+myClass = ExampleClass<String> new("foobar")
```

Comment:
`null` is a literal that is converted to a `Pointer`, thus `T` becomes `Pointer` instead of `String`. This bug is not really related to this bounty, but more related to the generic type system as a whole.